### DISCUSSION TITLE
timezones: IANA data versions

### DISCUSSION THREAD

**Discussion by calhob** [microsoft/STL#2468] [Q&A] [answered]
According to the standard, std::chrono::tzdb contains data that represents a copy of the IANA timezone database.
If I understand correctly, the stl implementation makes use of some icu.dll provided by the os. (I guess this is a custom build of the icu library dll made by Microsoft?)
As timezone rules tend to change (for example there have been 5 iana releases in 2021 (a-e)), will these changes become available through a reload_tzdb() call? Or are the rules completely embedded inside the icu.dll?
Is there any information on how many times the icu.dll will be updated so that it contains the new rules?
Or can you somehow instruct icu.dll to use your own download of the icu timezone data files?

On my system the tzdb version still seems to mention '2021a'.

**Comment by fsb4000** [comment, marked as answer]
I'm not a chrono expert, but here's my understanding:

> Is there any information on how many times the icu.dll will be updated so that it contains the new rules?

icu.dll updates come with Windows update.

> Or are the rules completely embedded inside the icu.dll?

Yes, I think so.

>  Or can you somehow instruct icu.dll to use your own download of the icu timezone data files?

I think you can't.

> On my system the tzdb version still seems to mention '2021a'.

ICU are not IANA data but they are close. And ICU is updated a little less often.

If you have some time you can watch 5 minutes from there for more information: https://youtu.be/AX0vTB1aQzI?t=2834
